### PR TITLE
Add Vitest

### PR DESCRIPTION
**Pull request by DanielKim1**
Set up Vitest, the recommended test runner for Nuxt (see https://nuxt.com/docs/getting-started/testing#features)

### CODE REVIEW

File: api/fetchImage.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-const axios = require("axios");
+import axios from "axios";
```
Comment (DanielKim1) [inline api/fetchImage.js:1]: > Vitest doesn't intercept require, so it is not possible to mock a module that was loaded with require

https://github.com/vitest-dev/vitest/discussions/3134#discussioncomment-5520134

**Comment by NatVIII** [comment]
Testing the functions of this website is something that it sorely lacks, I really appreciate any attempt to move towards better testing and fixing of features. Will work on approving now <3

**Review by NatVIII — CHANGES_REQUESTED** [review]
Rewrite the const and import stuff in api/fetchImage.js.

Images seem not to load in eventModals anymore, but I don't have an embedded way of knowing what type of error. It's very likely related to that change though.

**Comment by DanielKim1** [comment]
fetchImage changes in #10 will require different tests, will add tests in a new PR

**Pull request closed without merging**